### ISSUE TITLE
Unpublish back end links is not working

### ISSUE THREAD

**Issue by laoneo** [joomla/joomla-cms#13689]
### Steps to reproduce the issue
- On the back end open the menu manager.
- Select administrator
- Unpublish "Joomla! Update" in the list directly
![image](https://cloud.githubusercontent.com/assets/251072/22184637/22cb7214-e0d6-11e6-80bd-b14042161a48.png)

### Expected result
Joomla Update link in the back end menu under Components should not being visible anymore.

### Actual result
Link is visible
![image](https://cloud.githubusercontent.com/assets/251072/22184654/78d0df82-e0d6-11e6-9f1b-47c0a3ac660c.png)

### System information (as much as possible)
Latest staging.

**Comment by zero-24** [comment]
Did you set up a custom admin menu. And enabled it in the backend mod_menu?

**Comment by zero-24** [comment]
@izharaazmi can you take a look i'm also unable to understand and test this new feature in 3.7.0alpha2

**Comment by izharaazmi** [comment]
@laoneo @zero-24 Are you looking at the custom menu or the default preset menu? Unpublishing those menu items does not affect that in the preset menu. Please confirm.

**Comment by zero-24** [comment]
For the custom menu i have not found any way to change that menu or create a new one.

**Comment by izharaazmi** [comment]
You need to selected "System Links > Heading" in menu type and in the
second tab set that menu item's "Components Container" flag to yes. This
will cause the menu item to behave as the container for those menu items.

Complete details can be found in the actual PR.

**Comment by zero-24** [comment]
Ah. I was confused as the admin menu i have tryed to create just showed me the normal frontend links, yesterday. Today it looks like it work. But i guess @laoneo is come over the same issue than me.

**Comment by infograf768** [comment]
@izharaazmi
We need to find a way to clarify the use of this new feature. Maybe with a Notice when filtering by administrator client.

**Comment by izharaazmi** [comment]
@infograf768 What notice text do you suggest?

I think we should put everything in the user and developer docs. A link
with the HELP button should then be good. I am not sure where should I go
for this.


[image: View Izhar Aazmi's profile on LinkedIn]
<http://in.linkedin.com/in/izharaazmi> *Izharul Haque Aazmi (Izhar Aazmi)*
Bhartiy Digital Laboratories *and *Bhartiy Web Technologies,
11 - Kaveri Enclave - II, Near Swarn Jayanti Nagar, Ramghat Road, Aligarh -
202 001 (UP), India
Visit: *www.izharaazmi.com <http://www.izharaazmi.com>* :: Mail:
*izharaazmi@gmail.com
<izharaazmi@gmail.com>*

**Comment by laoneo** [comment]
First question, why can't I unpublish default menu items?

If you want to leave that behavior, then I suggest to disable the publish options in the menu items which can't be changed. Because the first thing I want to do when I set up a new site is to disable some default menu items, because I never use them. That would be so great to have that, to make Joomla slim in the back end, tailored to the needs of the site admin.

**Comment by izharaazmi** [comment]
> First question, why can't I unpublish default menu items?

You CAN. If you are using your custom admin menu. 
We just don't change the way the old preset behaves.

**Comment by infograf768** [comment]
Not sure, but what I am sure of is that something has to be done to prevent trashing/deleting Main (Protected) menu items.
In my opinion, we should not be able to change the status of protected menus.
In fact I think this is a release blocker.

I just made a new install where I could delete the Tags menu...
![screen shot 2017-01-24 at 14 43 33](https://cloud.githubusercontent.com/assets/869724/22249559/cf23c91c-e243-11e6-996e-0d53fd39c8d9.png)


![screen shot 2017-01-24 at 14 45 13](https://cloud.githubusercontent.com/assets/869724/22249573/db0eb674-e243-11e6-8426-278a70e934e2.png)

**Comment by izharaazmi** [comment]
Not being able to trash/delete - I do agree.

Not being able to unpublish - I don't think so. How else we would hide them
in our components menu container after we have created the relevant link at
our place of choice?

**Comment by infograf768** [comment]
Maybe a little more complex, but imho we should first check if a custom admin menu exists before being to publish/unpublish a protected menu item.
For trash delete, it is real important to correct now, maybe by showing the toolbar button ONLY when the list is filtered by site

**Comment by izharaazmi** [comment]
@infograf768 What is your opinion about batch operations on those protected
menu items?

**Comment by infograf768** [comment]
@izharaazmi 
Batch is also a useless one imho

**Comment by infograf768** [comment]
I tested this OK
```php
// Add a batch button
		if ($user->authorise('core.create', 'com_menus')
			&& $user->authorise('core.edit', 'com_menus')
			&& $user->authorise('core.edit.state', 'com_menus')
			&& $this->state->get('filter.client_id') == 0)
		{
			$title = JText::_('JTOOLBAR_BATCH');

			// Instantiate a new JLayoutFile instance and render the batch button
			$layout = new JLayoutFile('joomla.toolbar.batch');

			$dhtml = $layout->render(array('title' => $title));
			$bar->appendButton('Custom', $dhtml, 'batch');
		}

		if ($this->state->get('filter.published') == -2 
			&& $canDo->get('core.delete')
			&& $this->state->get('filter.client_id') == 0)
		{
			JToolbarHelper::deleteList('JGLOBAL_CONFIRM_DELETE', 'items.delete', 'JTOOLBAR_EMPTY_TRASH');
		}
		elseif ($canDo->get('core.edit.state') && $this->state->get('filter.client_id') == 0)
		{
			JToolbarHelper::trash('items.trash');
		}
```

**Comment by izharaazmi** [comment]
@infograf768 What I see in your code that you are hiding batch/delete/trash for ALL backend menu items. Is that what you intend?

IMHO this is not what we want for our custom menu items!

**Comment by infograf768** [comment]
@izharaazmi
Are you going to provide a patch?

**Comment by infograf768** [comment]
> What I see in your code that you are hiding batch/delete/trash for ALL backend menu items. Is that what you intend?

That is what I think makes the more sense and less prone to errors.

**Comment by infograf768** [comment]
Testing now your PR
https://github.com/joomla/joomla-cms/pull/13739

**Comment by izharaazmi** [comment]
> Are you going to provide a patch? 

I have already a minute ago 😄 

> That is what I think makes the more sense and less prone to errors.

How will we then move, copy, unpublish or delete a custom adminmenu  menu item?

**Comment by infograf768** [comment]
You are right

**Comment by laoneo** [comment]
I'v applied the patch. But I'm still able to unpublish menu items from the protected ones, which will appear in the back end menu as described in the issue description.

**Comment by infograf768** [comment]
Unpublishing these will ONLY work for a custom admin menu which contains a menu item of type Page Heading where the parameter "Component Menu Container" is set to yes.

Will comment in that PR for the rest

**Comment by izharaazmi** [comment]
> I'v applied the patch. But I'm still able to unpublish...

In the other PR? Unpublish is available on purpose. This is to let users hide those menu items from their custom menu items and still have a components menu container for any component they install later.

Use case is that they will hide all those and have a custom menu item where they desire. Therefore making the components container have nothing to show. This will cause that container to hide itself until a new component in installed with a new protected menu item for it.

**Comment by laoneo** [comment]
From my simple minded understanding, when I can unpublish a menu item in the protected menu then it should not being show when the protected menu is active. If this behavior should not being supported in the protected menu item, then it should be disabled. I have now a test environment with a unpublished menu item which is shown.

**Comment by izharaazmi** [comment]
@infograf768 What do you suggest about the behaviour of unpublished protected menu items for preset menu? @laoneo has a valid point. And what I did was under B/C consideration.

PS: Earlier (before this feature) those menu items were essentially unpublished and still always shown.

**Comment by infograf768** [comment]
Folks
@izharaazmi @laoneo 
Here are my thoughts:
https://github.com/joomla/joomla-cms/pull/13739#issuecomment-275032354

**Comment by franz-wohlkoenig** [comment]
@laoneo isn't this Issue fixed by #13830?

**Comment by laoneo** [comment]
I guess we can close this one or @izharaazmi and @infograf768 ?
